unitylibs/core/workflow/workflow-acrobat/action-binder.js

```diff
@@ -251,11 +272,21 @@ export default class ActionBinder {
       ),
     );
     await Promise.all(this.promiseStack)
-      .then((resArr) => {
+      .then(async (resArr) => {
         const response = resArr[resArr.length - 1];
         if (!response?.url) throw new Error('Error connecting to App');
         this.block.dispatchEvent(new CustomEvent(unityConfig.trackAnalyticsEvent, { detail: { event: 'redirect to product' } }));
-        window.location.href = response.url;
+        await this.waitForCookie(2000);
+        this.updateProgressBar(this.splashScreenEl, 100);
+        if (this.checkCookie()) {
+          window.location.href = response.url;
+        }
+        else {
+          await this.dispatchErrorToast('verb_cookie_not_set', 200, "Not all cookies found, redirecting anyway", true);
+          const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay));
+          await sleep(500);
+          window.location.href = response.url;
+        }
```

Comment:
It (L281-L289) looks like equivalent to the code below
```JS
        if (!this.checkCookie()) {
          await this.dispatchErrorToast('verb_cookie_not_set', 200, "Not all cookies found, redirecting anyway", true);
          await new Promise(r => setTimeout(r, 500));
        }
        window.location.href = response.url;
```

Reply:
Thanks @TsayAdobe I have incorporated suggested change.